client/landing/gutenboarding/onboarding-block/acquire-intent/vertical-select/index.tsx

```diff
@@ -184,7 +188,9 @@ const VerticalSelect: React.FunctionComponent< Props > = ( { onNext } ) => {
 
 	React.useEffect( () => {
 		inputRef.current.innerText = siteVertical?.label || '';
-	}, [ siteVertical, inputRef ] );
+		const isFreeForm = ! siteVertical?.slug && !! siteVertical?.label.length;
+		recordVerticalSelection( getSelectedVerticalUntranslatedLabel(), isFreeForm );
```

Comment:
I feel like the `getSelectedVerticalUntranslatedLabel()` selector doesn't really abstract much. Even though we're calling the selector here to hide the difference between slugs and labels, we're still having to deal with the nitty gritty of verticals in order to calculate the `isFreeForm` variable.

I can see that having a selector is useful for the `useTrackModal()` un-mount callbacks though. But maybe that's another reason why I think the slug and label should be two seperate props 😅

Reply:
You're right! Now if the user selects a vertical in French for example, we'll still send the non-translated, English label. But if the user's free-form input is a French phrase, we're sending that. So I guess it makes sense to record also the label.